### PR TITLE
Feat:lesson10

### PR DESCRIPTION
**Pull request by sae-github**
# [Lesson10](https://github.com/sae-github/handsonFrontend/blob/master/work/markup/1.md#10)

> 問題9に追加で try-catch-finaliy を書いてください

## [CodeSandBox](https://codesandbox.io/s/inspiring-galileo-xdgjb?file=/index.js) / a780134a276c91609e012b6bd999527630cd383a

[ comment ]
I was wondering which process to put in the "try."

It is implemented to receive the resolved data and catch it when an error occurs during the processing stage.

Please leave a review.

### CODE REVIEW

**Comment by sae-github** [comment]
> In this case, only 'getData' should be implemented in the ‘try', because if an error occurs, it is impossible to determine where the error occurred. (I was reviewed like this before!)

Thanks for the review :)
I have made the correction.
I am not sure if I am understanding your point correctly.

Please confirm.

cf29888bf995c6633f372c9fe2542a21d95b5c6f

**Comment by sae-github** [comment]
> 【IMO】
> it would be better to show some kind of message when an error occurs because nothing to show on the screen.
> 
> For example:
> catch (err) {
> ul.textContent = "データが取得できませんでした";
> console.error(err);
> }

I see...
I found it easy for users to understand.
I've used your code as is. Thank you very much.
cf29888bf995c6633f372c9fe2542a21d95b5c6f

**Review by csakyo — APPROVED** [review]

File: lesson10/index.js [hunk 1]
```diff
@@ -26,30 +27,33 @@ function getData() {
...
-    });
-    ul.appendChild(frag);
-  } catch (err) {
-    console.error(err);
-  }
-  finally {
+    return await getData();
+  } catch (e) {
+    console.error(e);
+    ul.textContent = "エラーが発生しました！";
```
Comment (nyattsu72) [inline lesson10/index.js]: (nist)

I'm sorry for the correction based on the review.

If it is ul.textContent, the text will be included in the <ul> tag.
I think it's better to put text other than the ul tag!

Comment (sae-github) [reply lesson10/index.js]: Thanks for the review :)
I understood that it is not a good idea to put text directly in the ul.
Fixed the code to get the div tag and insert the text inside the div when it catches an error.

Please confirm.
17b93893959ee33bfcddd8b75d2aa999182eec06

**Review by nyattsu72 — APPROVED** [review]

**Pull request merged**